dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/CommunityRestRepository.java

```diff
@@ -83,25 +81,45 @@ protected CommunityRest createAndReturn(Context context) throws AuthorizeExcepti
         }
 
         Community community;
+        try {
+            // top-level community
+            community = cs.create(null, context);
+            cs.update(context, community);
+            metadataConverter.setMetadata(context, community, communityRest.getMetadata());
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
 
+        return dsoConverter.convert(community);
+    }
+
+    @Override
+    @PreAuthorize("hasPermission(#id, 'COMMUNITY', 'ADD')")
+    protected CommunityRest createAndReturn(Context context, UUID id) throws AuthorizeException {
+
+        if (id == null) {
+            throw new DSpaceBadRequestException("The given parent parameter was invalid: "
+                + id);
+        }
 
+        HttpServletRequest req = getRequestService().getCurrentRequest().getHttpServletRequest();
+        ObjectMapper mapper = new ObjectMapper();
+        CommunityRest communityRest;
+        try {
+            ServletInputStream input = req.getInputStream();
+            communityRest = mapper.readValue(input, CommunityRest.class);
+        } catch (IOException e1) {
+            throw new UnprocessableEntityException("Error parsing request body: " + e1.toString());
```

Comment:
This is accidentally swallowing part of the exception and only returning the main error message (which is all the `toString()` will provide).  Sometimes that's "good enough", but other times, more info can be found from the entire stack trace.  I prefer to return the entire stacktrace like this:
`throw new UnprocessableEntityException("Error parsing request body", e1);`

Reply:
Done.  Updated the exception in CollectionRestRepository to also return the stack trace.